### PR TITLE
Send Inline 'auto_headers' when used 'with' Alien::OpenMP.

### LINKED ISSUE #10: define include header per compiler
The header (.h) file may differ based on compiler. For example `gcc`'s `libgomp` requires `#include <omp.h>`.

And not all compilers name this `omp.h`; this is also destined to work with `Inline::OpenMP` for injected the needful bits when inline'ing OpenMP based interfaces.

### PR DESCRIPTION
**Pull request by oodler577**

### CODE REVIEW

**Comment by oodler577** [comment]
I'll leave this up for a couple of days or until someone approves it. Please let me know if there are any things you think might be problematic.

File: t/02-with.t
Comment (kiwiroy) [inline t/02-with.t]: This is probably best also in `basic.t`, and also inverted, such as:

```suggestion
    is $config_ref->{AUTO_INCLUDE}, q{#include <omp.h>}, q{inspecting value of AUTO_INCLUDE.};
```

File: t/02-with.t
Comment (kiwiroy) [inline t/02-with.t:23]: Should this be a subtest with conditional skip on `$^O`?

File: lib/Alien/OpenMP.pm
Comment (kiwiroy) [inline lib/Alien/OpenMP.pm]: Plural or singular?

File: lib/Alien/OpenMP.pm
Comment (kiwiroy) [inline lib/Alien/OpenMP.pm]: Singular too.

File: alienfile
Comment (kiwiroy) [inline alienfile]: `auto_include` is not an attribute/property for `CBuilder` plugin.

To do, what I believe needs to happen, is augment the after `probe` block to be like and add an `auto_include` method to `Alien::OpenMP` (see below).
```perl
# only reached on success AFAICT
after probe => sub {
  my $build = shift;
  $build->install_prop->{'alien_openmp_compiler_has_openmp'} = 1;
  $build->runtime_prop->{auto_include}                       = Alien::OpenMP::configure->auto_include;
};
```

File: t/02-with.t
Comment (kiwiroy) [inline t/02-with.t]: Darwin fails this test.
```suggestion
    like $config_ref->{LDDLFLAGS},  qr/(?:-lomp|-fopenmp)/, q{inspecting value of LDDLFLAGS.};
```

File: lib/Alien/OpenMP.pm
Comment (kiwiroy) [inline lib/Alien/OpenMP.pm]:
```suggestion
    AUTO_INCLUDE  => $self->auto_include(), # e.g., #include <omp.h>
```

Also needs to define the method in this file, which is simply:
```perl
sub auto_include { shift->runtime_prop->{auto_include}; }
```

File: Changes
Comment (kiwiroy) [inline Changes]:
```suggestion
  to provide 'auto_include', which provides the
```

**Review by kiwiroy — CHANGES_REQUESTED** [review]
Looks good and makes for shorter codes.

Would be good to pick for consistency the use of singular vs plural `auto_include(s)` and bake into `alien.json` the value rather than _'configure'_ each time.

File: t/02-with.t
Comment (oodler577) [reply t/02-with.t:23]: Sure, whatt should the condition be? I don't have a great handle on that.

File: lib/Alien/OpenMP.pm
Comment (oodler577) [reply lib/Alien/OpenMP.pm]: good catch, singular

File: alienfile
Comment (oodler577) [reply alienfile]: mind a suggestion for this, I like how GH allows for this

Comment (kiwiroy) [reply alienfile]: I think the lines have to all be in the diff to allow for a suggestion to apply correctly.

File: lib/Alien/OpenMP.pm
Comment (kiwiroy) [reply lib/Alien/OpenMP.pm]:
```suggestion
    say Alien::OpenMP->auto_include;  # e.g. '#include <omp.h>' if gcc
```

File: t/02-with.t
Comment (kiwiroy) [reply t/02-with.t:23]:
```suggestion
subtest "Inline config" => sub {
    plan skip_all => "skip OS is Windows" if $^O eq "MSWin32";
```

**Comment by mohawk2** [comment]
Why is this removing the `non-linux` CI config?

**Comment by oodler577** [comment]
> Why is this removing the `non-linux` CI config?

Because I want to get the change in and not spend an inordinant amount of time worrying about CI; paricularly on a platform with a low chance of relevance, I'm happy to hack on the CI after tho.

**Comment by mohawk2** [comment]
> > Why is this removing the `non-linux` CI config?
> 
> Because I want to get the change in and not spend an inordinant amount of time worrying about CI; paricularly on a platform with a low chance of relevance, I'm happy to hack on the CI after tho.

Only just saw this. @zmughal has fixed up the problems, and I have added them on this branch instead of the now-unneeded CI removal. I believe he would be an excellent additional member of this GH org, by the way.

**Comment by mohawk2** [comment]
> @zmughal [...] I believe he would be an excellent additional member of this GH org, by the way.

I confused myself about the above - he is already a member, but I gather doesn't have push access to this repo for some reason.

File: t/02-with.t
Comment (zmughal) [reply t/02-with.t:23]: I made it skip on if `Inline::C` is available.

**Review by kiwiroy — APPROVED** [review]
LGTM

**Comment by oodler577** [comment]
> > @zmughal [...] I believe he would be an excellent additional member of this GH org, by the way.
> 
> I confused myself about the above - he is already a member, but I gather doesn't have push access to this repo for some reason.

He and @kiwiroy have been added with write perms. Thanks for pointing that out.

**Comment by oodler577** [comment]
Thanks @zmughal, @kiwiroy, @mohawk2 - this was a real team effort! Unless someone gets to it before me, I'll get this merge in and will make a release sometime this evening.

**Review by zmughal — APPROVED** [review]
👍 LGTM!

**Comment by mohawk2** [comment]
I have released this as 0.003006.

**Pull request merged**
